katdal/chunkstore_s3.py

```diff
@@ -14,35 +14,89 @@
 # limitations under the License.
 ################################################################################
 
-"""A store of chunks (i.e. N-dimensional arrays) based on the Amazon S3 API."""
+"""A store of chunks (i.e. N-dimensional arrays) based on the Amazon S3 API.
+
+It does not support S3 authentication/signatures, relying instead on external
+code to provide HTTP authentication.
+"""
 
 import contextlib
 import io
 import threading
 import Queue
 import sys
 import urlparse
+import urllib
+import hashlib
+import base64
+import warnings
+import contextlib
 
+import defusedxml.ElementTree
+import defusedxml.cElementTree
 import numpy as np
 try:
-    try:
-        import katsdpauth.auth_botocore   # noqa: F401
-    except ImportError:
-        import botocore
-    _botocore_import_error = None
+    import requests
+    from requests.adapters import HTTPAdapter as _HTTPAdapter
```

Comment:
I'm considering going the other direction, with everything optional so that you can build the katdal you want, e.g. `katdal[h5,ms,s3]` or `katdal[all]` like IPython.

Reply:
That makes some sense for packages that depend on C extensions since they can't always be installed easily, but I feel like the amount of time you're saving people in not downloading more packages (which for something like requests they probably have already) isn't justified given the much larger amount of time they'll lose by setting up a system, starting processing, and then getting an error when they realise they didn't install the optional dependencies.

Reply:
Absolutely - requiring requests and defusedxml seems pretty un-onerous...

Reply:
Point taken - I'll make `h5` and `ms` (and `rados`, poor `rados`) the only options then.